src/components/LoanCards/GridLoanCard.vue

```diff
@@ -88,31 +111,55 @@ export default {
 			}
 			return 'Expiring now!';
 		}
+	},
+	methods: {
+		toggleFavorite() {
+			// optimistically toggle it locally first
+			this.isFavorite = !this.isFavorite;
+
+			// @todo - does this need a .catch? if so, do we log error?
```

Comment:
You would only use `catch` for network errors. Any graphql errors are going to come through the `then` callback as part of the result. It can be accessed through destructuring like `data` is:
```.then({ data, errors }) => {```